### PR TITLE
kvserver: deflake TestLogGrowthWhenRefreshingPendingCommands

### LINKED ISSUE #89862: TestLogGrowthWhenRefreshingPendingCommands is flaky
This test failed 11 out of the last 1000 builds 

Screen Shot 2022-10-12 at 11 21 10 AM

------- Stdout: -------
=== RUN   TestLogGrowthWhenRefreshingPendingCommands
    test_log_scope.go:161: test logs captured to: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogGrowthWhenRefreshingPendingCommands607815779
    test_log_scope.go:79: use -show-logs to present logs inline
=== CONT  TestLogGrowthWhenRefreshingPendingCommands
    client_raft_test.go:1692: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestLogGrowthWhenRefreshingPendingCommands607815779
--- FAIL: TestLogGrowthWhenRefreshingPendingCommands (13.89s)

https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_UnitTests/6911410?showRootCauses=false&expandBuildChangesSection=true&expandBuildProblemsSection=true&expandBuildTestsSection=true

Jira issue: CRDB-20470

### PR DESCRIPTION
**Pull request by pav-kv**
Fixes #89862
    
Release justification: fixes flaky test
Release note: None

### CODE REVIEW

**Comment by pavelkalinnikov** [comment]
It looks like waiting on `WaitForLeaseUpgrade` is a cheap approach (but works to fix this test), in reality we need to be able to handle both cases. Another slightly longer fix would be to retry while we get `NotLeaseholder` errors.

File: pkg/kv/kvserver/client_raft_test.go [hunk 1]
```diff
@@ -1606,26 +1606,23 @@ func TestLogGrowthWhenRefreshingPendingCommands(t *testing.T) {
...
-			}
+			require.NoError(t, err)
 			repDesc, err := repl.GetReplicaDescriptor()
-			if err != nil {
-				t.Fatal(err)
-			}
-			if lease, _ := repl.GetLease(); !lease.Replica.Equal(repDesc) {
+			require.NoError(t, err)
+			lease, _ := repl.GetLease()
+			t.Logf("lease: %v", lease)
```
Comment (irfansharif) [inline pkg/kv/kvserver/client_raft_test.go:1624]: Assuming this was just for debugging, remove?

**Review by irfansharif — APPROVED** [review]
Nice, thanks! Don't forget to backport. (edit: didn't the see the label)

**Comment by pavelkalinnikov** [comment]
@irfansharif Backport to `22.2.x` or `22.2.0`?

**Comment by irfansharif** [comment]
.x I think? Whatever will be running a lot of CI and potentially observe CI flakes.

File: pkg/kv/kvserver/client_raft_test.go (hunk 1 above)
Comment (pavelkalinnikov) [reply pkg/kv/kvserver/client_raft_test.go]: Done.

**Comment by pavelkalinnikov** [comment]
bors r=irfansharif

**Comment by pavelkalinnikov** [comment]
bors r=irfansharif

**Pull request merged**